extensions/catppuccin/src/utils/icons.util.ts

```diff
@@ -0,0 +1,80 @@
+import { Image } from "@raycast/api";
+import { getFlavorPreference } from "./preferences.util";
+
+type ColorInput = string | undefined;
+
+const INVALID_ICONS = new Set([
+  "twitter",
+  "microsoftword",
+  "microsoftazure",
+  "microsoftpowerpoint",
+  "powershell",
+  "windowsterminal",
+  "visualstudio",
+  "visualstudiocode",
+]);
+
+const formatColor = (color: ColorInput = "#FFFFFF"): string => {
+  if (!color) return "ffffff";
+
+  if (typeof color === "string" && !color.includes("#")) {
+    const currentFlavor = getFlavorPreference();
+    const catppuccinColor = currentFlavor.colorEntries.find(([name]) => name.toLowerCase() === color.toLowerCase());
+    if (catppuccinColor) {
+      return catppuccinColor[1].hex.replace("#", "");
+    }
+  }
+
+  return color.replace("#", "");
+};
+
+export const getIcon = (iconName: string, color: ColorInput = "#FFFFFF"): Image.ImageLike => {
+  if (!iconName) {
+    return getDefaultIcon(color);
+  }
+
+  const normalizedIconName = iconName.toLowerCase();
+  const formattedColor = formatColor(color);
+
+  // check if icon ends .svg indiciating that it must be grabbed from the website repo
+  if (normalizedIconName.endsWith(".svg")) {
+    return {
+      source: `https://raw.githubusercontent.com/catppuccin/website/refs/heads/main/src/icons/ports/${normalizedIconName}`,
```

Comment:
The ref should be pinned to a hash here right?

Reply:
now using  69e2f3ee385279ea34e84c2c42703ed997eab40c